pkg/descheduler/pod/pods_test.go

```diff
@@ -74,6 +74,58 @@ func TestListPodsOnANode(t *testing.T) {
 	}
 }
 
+func TestGetPodOwnerReplicaSetReplicaCount(t *testing.T) {
+	t.SkipNow()
+	pod := test.BuildTestPod("pod1", 100, 0, "n1", nil)
+	replicasetName := "replicaset-1"
+	expectedReplicaCount := 3
+
+	test.BuildTestReplicaSet(replicasetName, 3)
+	ownerRef1 := test.GetReplicaSetOwnerRefList()
+	pod.ObjectMeta.OwnerReferences = ownerRef1
+
+	fakeClient := &fake.Clientset{}
+	fakeClient.Fake.AddReactor("get", "replicasets", func(action core.Action) (bool, runtime.Object, error) {
+		name := action.(core.GetAction)
+		fieldString := name.GetName()
+		if strings.Contains(fieldString, replicasetName) {
+			// how to return int as runtime.Object ? pod.DeepCopy is just a placeholder
+			return true, pod.DeepCopy(), nil
+		}
```

Comment:
Need help here. the reactor action expects an type that implements `DeepCopy`.
To return RS or RC, where do I define and implement the `DeepCopy` methods

Reply:
I think just rerutn the RS you created before using `test.BuildTestReplicaSet(replicasetName, 3)`, because `apps/v1/ReplicaSet` implements `runtime.Object`.

Reply:
ReplicationController hasnt implemented DeepCopy. Should I add it to this [file](https://github.com/kubernetes-sigs/descheduler/blob/61eef936183ecca959e34a4ba7940efb41b24a16/vendor/k8s.io/api/apps/v1/zz_generated.deepcopy.go)

`fakeClient.Fake.AddReactor("get", "replicationcontrollers", func(action core.Action) (bool, runtime.Object, error) {` - Is this syntax **equivalent of faking** `client.AppsV1().ReplicaSets(v1.NamespaceAll).Get(ctx, ownerRef.Name, metav1.GetOptions{})`

Reply:
RC is under core/v1 and here is its [DeepCopy](https://github.com/kubernetes-sigs/descheduler/blob/61eef936183ecca959e34a4ba7940efb41b24a16/vendor/k8s.io/api/core/v1/zz_generated.deepcopy.go#L4260), besides, we shouldn't change autogenerated and vendor files. You can reference to [this project ](https://github.com/kubernetes/code-generator) which is used to generate these files.